packages/server/src/transform/mpegts.ts

```diff
@@ -0,0 +1,185 @@
+  return new UnixStream(stream, id, (socket) => {
+    stream.on("error", (err) => {
+      console.log(`INPUT(${id}):` + err.message);
+    });
+    socket.on("error", (err) => {
+      console.log(`OUTPUT(${id}):` + err.message);
+    })
+    stream.pipe(socket);
+  });
+}
+
+var MPEGTS_INTER_PORT = 9000;
+
+export class MPEGTS {
+  private pc: RTCPeerConnection;
+  private interPort: number;
+  private streams;
+  private id: string;
+
+  constructor(pc: RTCPeerConnection, id) {
+    this.pc = pc;
+    this.interPort = MPEGTS_INTER_PORT++;
+    this.id = id;
+
+    let videoSink;
+    let audioSink;
+
+    this.pc.getReceivers().forEach(({ track }) => {
+      if (track.kind === "video") {
+        videoSink = new RTCVideoSink(track);
+      } else if(track.kind === "audio") {
+        audioSink = new RTCAudioSink(track);
+      }
+    });
+
+    this.streams = [];
+
+    videoSink.addEventListener("frame", ({ frame: { width, height, data }}) => {
+      const streamId = width + "x" + height;
+      if (!this.streams[0] || (this.streams[0] && this.streams[0].streamId !== streamId)) {
+        // A new frame size has arrived we need to create a new stream and stop
+        // the others.
+        const stream = this.createStream(streamId, audioSink);
+        this.streams.unshift(stream);
+        this.streams.forEach(item => {
+          if (item !== stream && !item.end) {
+            item.end = true;
+            if (item.audio) {
+              item.audio.end();
+            }
+            item.video.end();
+          }
+        });
+        stream.proc.run();
+      }
+      this.streams[0].video.push(Buffer.from(data));
+    });
+
+    const { close } = this.pc;
+    this.pc.close = () => {
+      audioSink.stop();
+      videoSink.stop();
+      this.streams.forEach(({ audio, video, end, }) => {
+        if (!end) {
+          if (audio) {
+            audio.end();
+          }
+          video.end();
+        }
+      });
+      return close.apply(this, arguments);
+    }
+  }
+
+  createStream(streamId, audioSink) {
+    const stream = {
+      streamId: streamId,
+      video: new PassThrough(),
+      audio: new PassThrough(),
+      end: false,
+      proc: null,
+    };
+
+    const onAudioData = ({ samples: { buffer }}) => {
+      if (!stream.end) {
+        stream.audio.push(Buffer.from(buffer));
+      }
+    };
+
+    audioSink.addEventListener("data", onAudioData);
+
+    stream.audio.on("end", () => {
+      audioSink.removeEventListener("data", onAudioData);
+    });
+    stream.audio.on("error", (err) => {
+      console.log("A:" + err.message);
+    });
+
+    stream.video.on("error", (err) => {
+      console.log("V:" + err.message);
+    });
+    
+    stream.proc = ffmpeg()
+      .addInput(StreamInput(stream.video, this.getId() + `-${stream.streamId}-v`).url)
+      .addInputOptions([
+        "-f", "rawvideo",
+        "-pix_fmt", "yuv420p",
+        "-s", stream.streamId,
+        "-r", "30",
```

Comment:
should this really be hardcoded?

Reply:
Yes, I believe this is what wrtc does write on the sink